File: GXpert/GXpert.Web/Modules/Masters/Taluka/TalukaColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.Taluka")]
[BasedOnRow(typeof(TalukaRow), CheckNames = true)]
public class TalukaColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    [QuickFilter]
    public string StateTitle { get; set; }
    [QuickFilter(CssClass = "hidden-xs"), QuickFilterOption("cascadeFrom", "StateId")]
   
    public string DistrictTitle { get; set; }
    public string ShortName { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/TalukaColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TalukaRow } from "./TalukaRow";

export interface TalukaColumns {
    Id: Column<TalukaRow>;
    Title: Column<TalukaRow>;
    StateTitle: Column<TalukaRow>;
    DistrictTitle: Column<TalukaRow>;
    ShortName: Column<TalukaRow>;
}

export class TalukaColumns extends ColumnsBase<TalukaRow> {
    static readonly columnsKey = 'Masters.Taluka';
    static readonly Fields = fieldsProxy<TalukaColumns>();
}